File: src/main/java/com/gitlab/daring/image/command/parameter/CommandParamUtils.java
package com.gitlab.daring.image.command.parameter;

import one.util.streamex.StreamEx;

import java.util.Collection;
import java.util.Map;

public class CommandParamUtils {

    public static Map<String, Object> buildParamConfig(Collection<CommandParam<?>> params) {
        return StreamEx.of(params)
            .filter(p -> p.configPath != null)
            .toMap(p -> p.configPath, CommandParam::configValue);
    }

    private CommandParamUtils() {
    }
}

File: src/main/kotlin/com/gitlab/daring/image/command/parameter/CommandParamUtils.kt
package com.gitlab.daring.image.command.parameter

object CommandParamUtils {

    @JvmStatic
    fun buildParamConfig(params: Collection<CommandParam<*>>): Map<String, Any> {
        return params.filter { it.configPath != "" }
                .associateBy( {it.configPath }, CommandParam<*>::configValue)
    }

}